drud/ddev
# Use sentry to log usage events, fixes #307

## The Problem/Issue/Bug:

* We want opt-in stats and crash reports for ddev OP #307 

## How this PR Solves The Problem:

* Add global configuration (~/.ddev/global_config.yaml)
* Add `ddev config global` as command for people to change global config
* Add opt-in prompt
* Integrate with Sentry for most commands (reports the command and environment) and also for panics and the like. 
* Add docker and docker-compose version to `ddev version`
* `ddev version` now presents sorted, stable list of values.

## Manual Testing Instructions:

* This must be built either by CircleCI or by make with `export SentryDSN=<dsn_value>` set
* Get SentryDSN at https://sentry.io/settings/drud/ddev/keys/
* Do some things, create some errors
* Visit the ddev dashboard on Sentry (you can log in with github credentials)
* Validate what you find there.

Misc:
* Remove ~/.ddev/global_config.yaml and verify expected prompting behavior'
* Change last_run_version there to verify expected prompting behavior
* Set omit_containers to omit dba and ddev-ssh-agent in the global config, verify behavior.
* Look at `ddev version` and verify content
* Experiment with `ddev config global`, introducing errors, etc.

## Automated Testing Overview:

Still needs tests

## Related Issue Link(s):

OP #307

<img width="884" alt="ddev_rm" src="https://user-images.githubusercontent.com/112444/49121518-f88ac080-f26d-11e8-9e27-8588720fcb01.png">
